### ISSUE TITLE
Carbons not working

### ISSUE THREAD

**Issue by sutr90** [gkdr/carbons#5]
I'm not sure if I'm using this correctly, so this might not be an issue, but we'll see.

Use case:

> _Pidgin Win10, latest ( 2017-02-16 12:36 ) version of carbons_
> [me 10:00] /carbons on
> **Turning carbons ON for <user>
> Success!**
> ~~~~ 10:30 ~~~~
> [me 10:30] sending test message
> [peer@ubuntu-jabber.de 10:30] fine I got your message
> 
> Pidgin Ubuntu, latest ( 2136d9e ) version of carbons
> [me 10:10] /carbons on
> **Turning carbons ON for <user>
> Success!**
> ~~~~ 11:00 ~~~~
> [me 11:00] well, it seems it is not working

As you can see, even though the carbons are enabled and the server reports it is working, I don't receive the messages on the second client.

**Comment by mightyBroccoli** [comment]
Are you sure that the server is supporting carbons?

**Comment by sutr90** [comment]
Not really. I though that it is working if the plugin says so. I will send email to the server admins, to check.

**Comment by gkdr** [comment]
@mightyBroccoli The 'Success!' message is only displayed if the server returns a positive response, so that should not be the issue.

@sutr90 You do seem to be using it correctly, so I'm going to need some debug output. You can get it by going to _Help -> Debug Window_. There you can see __carbons__ telling you that it ignores stuff (probably). I need the raw XML that __jabber__ is writing to the log right before (or maybe right after).

**Comment by EionRobb** [comment]
@sutr90 did you build the windows dll yourself?  how do you know which revision it was built from?

**Comment by sutr90** [comment]
Here is the output to debug window of the Pidgin startup on Ubuntu:
[purple-debug.txt](https://github.com/gkdr/carbons/files/830346/purple-debug.txt)

At line 58 I can see: `<feature var='urn:xmpp:carbons:2'/>` so I guess the carbons should be enabled.
PS: I have redacted the peers info, I guess it should not be a problem.

@EionRobb good catch. I have used the Windows version from your server:
carbons.dll	2017-02-16 12:36 	67K	 
So it is actually older. I will update the issue.

**Comment by gkdr** [comment]
Sorry, I should have clarified. Well, this is good too, now I know that the server replies it is working.
But I need you to send a message from the Windows build to someone, and then look what happens in the debug log of the Ubuntu build (from the debug branch).

**Comment by sutr90** [comment]
Okay, here is the new log.
[purple-debug.txt](https://github.com/gkdr/carbons/files/830971/purple-debug.txt)

I have sent some msgs from Windows, they were received by the peer. Then I have logged off in Windows, switched to Ubuntu with carbon debug build and logged in the pidgin.

**Comment by gkdr** [comment]
I have another idea - I didn't add carbons to the "entity capabilities", partly because I couldn't find the right function (but maybe also because you can't do that while the account is connected, I can't quite remember).
Maybe the ejabberd implementation does not send the messages because it can't find carbons in the "caps". I will change this so you can try again, but for today I ran out of time.

**Comment by sutr90** [comment]
No problem!

I really appreciate your support and all the effort which goes to this. 
Thanks.

**Comment by gkdr** [comment]
I am now adding carbons to the list of features (like I was supposed to).
Can you please pull the newest commit from the debug branch (https://github.com/gkdr/carbons/commit/7ed0746ee85980cee66748bfc93586caead0c8e0) and tell me if it improved for you?

**Comment by sutr90** [comment]
Hello, 
this is really embarrasing and I'm terribly sorry. :(

I was using the plugin incorrectly the whole time. No when testing the latest debug version I have noticed the updated description of the plugin in the latest version (7ed0746). The relevant part  
> which are **[online]** at the same time.

That's the reason I thought it is not working. I sent some messages from ComputerA, turned it off. And logged in on Computer B. And I was expecting to get the messages. Which is not how this is supposed to work!

In the end I have tested both debug and master branches, and both of them work perfectly.
